_opensearch/install/plugins.md

```diff
@@ -7,61 +7,173 @@ nav_order: 90
+
+#### Usage:
 ```bash
 bin/opensearch-plugin install <plugin-name>
 ```
 
-For a plugin installed via zip, use:
+#### Example:
 ```bash
-bin/opensearch-plugin install <name|Zip File|Url>
+$ sudo ./opensearch-plugin install analysis-icu
+-> Installing analysis-icu
+-> Downloading analysis-icu from opensearch
+[=================================================] 100%   
+-> Installed analysis-icu with folder name analysis-icu
 ```
 
-Restart your OpenSearch node after installing a plugin.
+### Install a plugin from a zip file:
 
-## Batch mode
+#### Usage:
+```bash
+# zip-file can be a local path to the zip file, or
+# a URL for a zip file hosted on a network.
+bin/opensearch-plugin install <zip-file>
+```
 
-When installing plugins that require additional privileges not included by default, the plugins will prompt the user for confirmation of the required privileges. To grant all requested privileges, use batch mode to skip the confirmation prompt.
+#### Example:
+```bash
+$ sudo ./opensearch-plugin install https://repo1.maven.org/maven2/org/opensearch/plugin/opensearch-anomaly-detection/2.2.0.0/opensearch-anomaly-detection-2.2.0.0.zip
+-> Installing https://repo1.maven.org/maven2/org/opensearch/plugin/opensearch-anomaly-detection/2.2.0.0/opensearch-anomaly-detection-2.2.0.0.zip
+-> Downloading https://repo1.maven.org/maven2/org/opensearch/plugin/opensearch-anomaly-detection/2.2.0.0/opensearch-anomaly-detection-2.2.0.0.zip
+[=================================================] 100%   
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@     WARNING: plugin requires additional permissions     @
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+* java.lang.RuntimePermission accessClassInPackage.sun.misc
+* java.lang.RuntimePermission accessDeclaredMembers
+* java.lang.RuntimePermission getClassLoader
+* java.lang.RuntimePermission setContextClassLoader
+* java.lang.reflect.ReflectPermission suppressAccessChecks
+* java.net.SocketPermission * connect,resolve
+* javax.management.MBeanPermission org.apache.commons.pool2.impl.GenericObjectPool#-[org.apache.commons.pool2:name=pool,type=GenericObjectPool] registerMBean
+* javax.management.MBeanPermission org.apache.commons.pool2.impl.GenericObjectPool#-[org.apache.commons.pool2:name=pool,type=GenericObjectPool] unregisterMBean
+* javax.management.MBeanServerPermission createMBeanServer
+* javax.management.MBeanTrustPermission register
+See http://docs.oracle.com/javase/8/docs/technotes/guides/security/permissions.html
+for descriptions of what these permissions allow and the associated risks.
+
+Continue with installation? [y/N]y
+-> Installed opensearch-anomaly-detection with folder name opensearch-anomaly-detection
+```
 
-To force batch mode when installing plugins, add the `-b` or `--batch` option:
+### Install a plugin using Maven coordinates:
```

Comment:
it's maybe important to add a note about the restriction that this only goes against maven central (it doesn't actually use maven to find the artefact but it just constructs the URL in the background), so there's (currently) no way to e.g. configure it to go against a proxy maven server instead (as you could do with normal maven builds).

it can well be that in a build environment there's no access to the internet but there is access to internal mirrors. accordingly, one might think that with this method it'd be possible to access it from the maven mirror (because the proper maven setting is defined for the user running the command).

Reply:
Thanks for calling that out!  I'll open another issue for that since this one was already merged.

Reply:
Opened up https://github.com/opensearch-project/documentation-website/issues/1051 for tracking